開発環境のデータベースをMySQLからPostgreSQLに変更

toshinori-m opened the pull request:
### 概要
本番環境でPostgreSQLを使用しているため、開発環境・テストもPostgreSQLに統一し、環境間の違いによる問題を防ぐことを目的としています。
### 変更内容
- `database.yml`の`default`を`PostgreSQL`用にしました。
- `docker-compose.yml`の`db`で`PostgreSQL`を使用するようにしました。
- `Gemfile`から開発環境・テストで使用していた`mysql2`を削除し、全ての環境で`PostgreSQL`を使用するようにしました。
### 確認内容
- ローカル環境の`Docker`コンテナ内で以下を確認しました。
  1. docker compose down
  2. docker compose run --rm backend bash -c "bundle exec rails db:create && bundle exec rails db:migrate && bundle exec rails db:seed"
  3. docker compose build　→ docker compose up -d
  4. docker compose exec db psql -U postgres -d myapp_development　⇨（エラーなく実行）

close https://github.com/toshinori-m/stay_connect/issues/133

hirosi1900day reviewed (approved):
[ここも](https://github.com/toshinori-m/stay_connect/blob/04aafdf483eb75d3987493928843842e27b7e9b7/backend/config/database.yml#L54)指定は不要そう

それ以外は問題なし

toshinori-m committed:
開発環境のデータベースをMySQLからPostgreSQLに変更

- database.ymlをPostgreSQL用に修正
- docker-compose.ymlでPostgreSQLを使用するよう設定を変更
- Gemfileからmysql2を削除し、pgを追加
- PostgreSQLの接続確認およびセットアップ手順を更新
